cloudfoundry-operations/src/main/lombok/org/cloudfoundry/operations/domains/ListDomainsRequest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.cloudfoundry.operations.domains;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Singular;
+import lombok.ToString;
+import org.cloudfoundry.Validatable;
+import org.cloudfoundry.ValidationResult;
+import org.cloudfoundry.client.v2.InFilterParameter;
+import org.cloudfoundry.client.v2.PaginatedRequest;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public final class ListDomainsRequest extends PaginatedRequest implements Validatable {
```

Comment:
@LokeshN Based on the inputs to the CLI

```
$: cf help domains
NAME:
   domains - List domains in the target org

USAGE:
   cf domains
```

There is no need for a request object.  And example of this would be [`Organizations.list()` operation](https://github.com/cloudfoundry/cf-java-client/blob/master/cloudfoundry-operations/src/main/java/org/cloudfoundry/operations/organizations/Organizations.java#L56).